R/derive_param_bor.R

```diff
@@ -0,0 +1,541 @@
+#'
+#'    *Required or Optional:* Required
+#'
+#' @param filter_source Filter to be applied to `dataset` to derive the
+#'                       Best Overall Response
+#'
+#'    *Required or Optional:* Required
+#'
+#' @param source_pd Date of first progressive disease (PD)
+#'
+#'   If the parameter is specified, the observations of the input `dataset` for
+#'   deriving the new parameter are restricted to observations up to the
+#'   specified date. Observations at the specified date are included. For
+#'   subjects without first PD date all observations are take into account.
+#'
+#'   *Permitted Values:* a `date_source` object (see `admiral::date_source()`
+#'   for details)
+#'
+#'   *Default:* `NULL`
+#'
+#'   *Required or Optional:* Optional
+#'
+#' @param source_datasets Source dataframe to be used to calculate the
+#'                        first PD date
+#'
+#'   A named list of dataframes is expected (although for BoR) only one dataframe is
+#'   needed. It links the `dataset_name` from `source_pd` with an existing dataframe.
+#'
+#'   For example if `source_pd = pd_date` with
+#'   ```
+#'   pd_date <- date_source(
+#'     dataset_name = "adrs",
+#'     date = ADT,
+#'     filter = PARAMCD == PD
+#'   )
+#'   ```
+#'   and the actual response dataframe in the script is `myadrs`, `source_datasets
+#'   = list(adrs = myadrs)` should be specified.
+#'
+#'    *Required or Optional:* Optional
+#'
+#' @param reference_date Reference date
+#'
+#'   The reference date is used along with `ref_start_window` to determine those
+#'   records that occur before and after `ADT` (see Details section for further
+#'   information). Usually it is treatment start date (`TRTSDT`) or
+#'   randomization date (`RANDDT`).
+#'
+#'   *Permitted Values:* a numeric date column
+#'
+#'   *Required or Optional:* Required
+#'
+#' @param ref_start_window Stable disease time window
+#'
+#'   The ref_start_window is used along with `reference_date` to determine those
+#'   records that occur before and after `ADT` (see Details section for further
+#'   information).
+#'
+#'   *Permitted Values:* a non-negative numeric scalar
+#'
+#'   *Default:* 0
+#'
+#'   *Required or Optional:* Optional
+#'
+#' @param missing_as_ne Consider no assessments as `"NE"`?
+#'
+#'   If the argument is set to `TRUE`, the response is set to `"NE"` for
+#'   subjects in `dataset_adsl` without an assessment in the `dataset` after the
+#'   filter has been applied. Otherwise, the response is set to `"MISSING"`
+#'   for these subjects.
+#'
+#'   *Permitted Values:* a logical scalar
+#'
+#'   *Default:* `FALSE`
+#'
+#'   *Required or Optional:* Required
+#'
+#' @param aval_fun Function to map character analysis value (`AVALC`) to numeric
+#'                 analysis value (`AVAL`)
+#'
+#'   The (first) argument of the function must expect a character vector and the
+#'   function must return a numeric vector.
+#'
+#'   *Default:* `admiralonc::aval_resp` (see `?admiralonc::aval_resp()`)
```

Comment:
admiralonco is the package name not admiralonc

Reply:
all package prefixes removed, except in testing.